Docker compose

ronja-koistinen opened the pull request:
Work in progress, do not merge. Almost works.

ronja-koistinen committed:
Local dev environment on Docker compose

This introduces a local development environment base on Docker compose.

Host dependencies:
* docker (for container runtime)
* docker-compose (for container orchestration)
* docker-xbuild (for image builds)
* socat (for an ssh-agent hack)

See the comments in ./dev-docker.sh for further information.

ronja-koistinen commented:
Nyt tämä toimii. Tietokannan alustaminen on nyt edelleen mariadb-kontin homma, koska siinä on niin kivat valmiit skriptit, ja niiden toteuttaminen ad hoc uudelleen vufind-kontin puolelle olisi aika työlästä pienen pedanttisen tyyliseikan vuoksi...

ronja-koistinen committed:
Fix config numbering, remove commented noise

EreMaijala commented on local/docker/xdebug.ini:
```diff
@@ -0,0 +1,13 @@
+memory_limit = 2048M
```
Tämä tiedosto sisältää muutakin kuin XDebugin asetukset, joten nimi on vähän harhaanjohtava.

EreMaijala reviewed (changes requested):
Näyttää ihan hyvältä. Oikeastaan ainoa isompi huoli on se, että jos joku keksii lisätä docker compose -tuen yläjuoksulle, niin siitä tulee konflikteja, mutta voidaan selvitellä asiaa tarpeen mukaan, jos se tulee eteen.

ronja-koistinen committed:
Remove crud from ini file, rename it

ronja-koistinen committed:
Streamline SSH proxy config and obfuscate examples

EreMaijala commented:
@ronja-koistinen Sittenkin vielä yksi kysymys: olisiko parempi laittaa local/config:n alla olevat uudet tiedostot docker-hakemiston alle? Ne kun eivät ihan suoraan liity VuFindin konfiguraatioon ja voi tulla myöhemmin vähän hämmennystä, että miksi on vaikka tosi turvaton tunnus tuolla grants.sql:ssä.

ronja-koistinen committed:
More changes

- Move all Docker env specific config files to local/docker
- Rename php ini file again
- Add increased memory_limit to php-localdev.ini

EreMaijala commented on dev-docker.sh:
```diff
@@ -0,0 +1,47 @@
+# Install socat, docker, docker-compose and docker buildx.
+#
+# For configuration, fill in:
+#   * .env
```
```suggestion
#   * .env (see template in env.sample)
```

EreMaijala commented on env.sample:
```diff
@@ -0,0 +1,15 @@
+FINNA_DOCUMENT_ROOT=/opt/vufind2
```
```suggestion
# Environment variables for Docker

FINNA_DOCUMENT_ROOT=/opt/vufind2
```

ronja-koistinen committed:
Apply suggestions from code review

Co-authored-by: Ere Maijala <ere.maijala@helsinki.fi>
